### PR TITLE
Core

### PR DESCRIPTION
**Pull request by jonnystening**
Implementing global css resets and typography styling in the core component

Styling has been separated into resets, type, fonts and extensions for clarity; deep merged via lodash.merge

Basic text styling on the example App (`helpers/example/index.js`) has been removed

Note: This is a WIP. Refer to the component's ticket for more... _see_ https://github.com/WestpacGEL/GEL/issues/51

### CODE REVIEW

File: brands/BOM/tokens/index.js [hunk 1]
```diff
@@ -63,7 +98,71 @@ export default {
 			foreground: 'white',
 		},
 	},
-	type: {},
+	font: FONT(),
+	typography: {
+		fontSize: 10,
```
Comment (florettewest) [inline brands/BOM/tokens/index.js]: The base font size should be set in scalable unit (em or rem)

[the remark this answers is not in the document]
Comment (jonnystening) [reply brands/BOM/tokens/index.js]: There's a bit of debate about this. Latest method seems to be to use fixed px for base reset and relative rems in the body. 

I'll see if I can find the articles I was referring to.

File: brands/BOM/tokens/index.js [hunk 2]
```diff
@@ -273,6 +372,70 @@ export default {
...
+			},
+			invalid: {
+				borderColor: COLORS.danger,
+			},
+		},
+		size: {
+			small: {
+				padding: ['3px', '9px', '4px'],
+				fontSize: '14px',
+				height: '30px',
```
[the remark this answers is not in the document]
Comment (jonnystening) [reply brands/BOM/tokens/index.js]: Need to discuss. Will update this on the form-input branch.

File: components/core/examples/10-fonts.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		<p className="body-font" style={{ fontWeight: 600 }}>
+			SemiBold: 600
+		</p>
+		<p className="body-font" style={{ fontWeight: 700 }}>
+			Bold: 700
+		</p>
+		<p className="body-font" style={{ fontWeight: 800 }}>
+			Black: 800
+		</p>
+		<p className="body-font" style={{ fontWeight: 900 }}>
```
[the remark this answers is not in the document]
Comment (jonnystening) [reply components/core/examples/10-fonts.js]: It depends on what font the user's machine decides to use. These are just here to ensure we're checking all on all devices.

File: brands/BOM/tokens/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (florettewest) [reply brands/BOM/tokens/index.js]: If you use px as a base reset, you are blocking the customisable font size in the browser setting: no good for accessibility!

[the remark this answers is not in the document]
Comment (jonnystening) [reply brands/BOM/tokens/index.js]: Yeah I'm across this too. Not sure how what I was reading was supposed to work, it certainly doesn't resize text with a user font size adjustment 🤔. Reckon they were just ignoring these users. Not good.

css-tricks.com sets font-size: 20px for the document. Keen to know Chris' reasoning on this..

So we need to decide if we really need to have our 10px reset (using relative units). It makes rem calculations easier, but may have some drawbacks.

html { font-size: 62.5%; } /* 10px */
body { font-size: 1.4rem; } /* 14px */

or

html { font-size: 100%; } /* or not set at all, prob better */
body { font-size: 0.875rem } /* 14px */

The latter obviously means we will have a harder time working with rems, but is probably more correct for users.

File: brands/BOM/tokens/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonnystening) [reply brands/BOM/tokens/index.js]: I've removed the height token, we can just calc this like how Bootstrap does it, and moved padding and fontSize to rems

File: components/form-input/src/FormInput.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
+/** @jsx jsx */
```
Comment (jaortiz) [inline components/form-input/src/FormInput.js]: Please be careful not to bring in other components that are not relevant to this pr.

File: components/core/src/Core.js [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+			...typography.link.default,
+
+			':hover': {
+				...typography.link.hover,
+			},
+		},
+	};
+
+	// Font @fontface styling
+	const styleFont = font;
```
Comment (jaortiz) [inline components/core/src/Core.js]: unnecessary, just use the font variable as is

File: components/core/src/Core.js [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+import { paint } from './utils';
+
+import merge from 'lodash.merge';
+
+export const Core = ({ children }) => {
+	const { colors, breakpoints, font, typography } = useTheme();
+	const isKeyboardUser = useIsKeyboardUser();
+	const mq = paint(breakpoints);
+
+	console.log(`core: ${isKeyboardUser}`);
```
Comment (jaortiz) [inline components/core/src/Core.js]: remove console.logs

File: components/core/src/Theme.js [hunk 7]
```diff
@@ -19,21 +22,47 @@ export const useTheme = () => {
 	return themeObject;
 };
 
-export const GEL = ({ brand, ...props }) => {
+export const useIsKeyboardUser = () => {
```
Comment (jaortiz) [inline components/core/src/Theme.js]: update keyboardUser stuff as discussed

File: components/form-input/src/FormInput.js (hunk 4 above)
Comment (jonnystening) [reply components/form-input/src/FormInput.js]: will remove them 👍

Comment (jonnystening) [reply components/form-input/src/FormInput.js]: I'll remove it for now so we can merge `core-styling` into `gel-develop`. But will have to hook this back up once we merge `form` into `gel-develop`.

File: components/core/src/Core.js (hunk 5 above)
Comment (jonnystening) [reply components/core/src/Core.js]: ok, will remove 'style' prefix from all emotion passed css object going forward

Comment (jaortiz) [reply components/core/src/Core.js]: I'm fine with the rename in the destructure

**Pull request merged**